### ISSUE TITLE
Heather reported issue: EasyCLA - Not all Signed ICLAs are showing in table

### ISSUE THREAD

**Issue by vinod-kadam** [communitybridge/easycla#2702]
In the EasyCLA overview section, the table does not list all ICLAs signed for the project.

Example: OpenVDB
The PCC table shows 7 ICLA signatures.
The project console v1 shows 20 ICLA signatures
Verified in the database there are 20 signatures for the project where signature_reference_type = user, signature_signed = true, and signature_user_ccla_company_id is empty.

Tejas investigated it: I Verified the PROD API https://api-gw.platform.linuxfoundation.org/cla-service/v4/api-docs#operation/listClaGroupIclaSignature with CLA group ID "8034cd49-0799-4fc6-9f4f-27b8a75cfdb1" its returning 7 records from Backend. So this seems to be a backend issue.

Note: Original ticket created by Heather was https://jira.linuxfoundation.org/browse/LFX-3120 however as Tejas investigated it and found that its a backend issue so created this ticket.

**Comment by dealako** [comment]
Investigating - for this CLA Group - we should return a total of 20 ICLA signatures that have been signed/approved:

![Screen Shot 2021-02-25 at 11.52.21 AM.png](https://images.zenhubusercontent.com/5ed6694b9d49d1a6a85690e0/4a49f7bd-a975-4df1-a50c-3b6e3f2b7d69)

**Comment by dealako** [comment]
Updated logic to apply the correct ICLA signature query.  Augmented the results with missing data.  A number of records excluded the github username and/or the user's name/emails address.  These values are now fetched and added to the response when available.

```
{
    "list": [
        {
            "github_username": "<redacted>",
            "signatureModified": "2020-10-02T09:18:01.735025+0000",
            "signature_id": "784bebee-29bc-4de6-86af-40b737148a83",
            "signed_on": "2020-10-02T09:35:32Z",
            "user_email": "antoine.<redacted>@<redacted>.fr",
            "user_name": "Antoine <redacted>"
        },
        {
            "github_username": "<redacted>",
            "signatureModified": "2020-02-21T19:13:07.060843+0000",
            "signature_id": "1ed2404d-2a3f-4697-b382-1da3daa9d373",
            "signed_on": "2020-02-21T19:13:07.060836+0000",
            "user_email": "<redacted>@gmail.com"
        },
        {
            "github_username": "<redacted>",
            "signatureModified": "2020-10-30T18:58:01.985357+0000",
            "signature_id": "b625f576-e6a0-4bb6-82ac-f3ba9fb90fba",
            "signed_on": "2020-10-30T18:59:14Z",
            "user_email": "<redacted>@mac.com",
            "user_name": "Jamie <redacted>"
        },
        {
            "github_username": "<redacted>",
            "signatureModified": "2020-05-25T06:48:33.790079+0000",
            "signature_id": "ddd68752-20c1-45f6-84bd-5d960698a755",
            "signed_on": "2020-05-25T06:48:33.790072+0000",
            "user_email": "<redacted>@hotmail.com",
            "user_name": "<redacted> <redacted>"
        },
        {
            "github_username": "<redacted>",
            "signatureModified": "2020-03-27T19:16:09.201005+0000",
            "signature_id": "19cd032f-2229-40a5-9212-9d5cd4c45662",
            "signed_on": "2020-03-27T19:16:09.200997+0000",
            "user_email": "<redacted>@cs.<redacted>.edu",
            "user_name": "Michael <redacted>"
        },
        {
            "github_username": "<redacted>",
            "signatureModified": "2020-03-18T09:39:24.419998+0000",
            "signature_id": "34d612af-4dcf-4240-83ff-54bdf8641435",
            "signed_on": "2020-03-18T09:39:24.419990+0000",
            "user_email": "<redacted>@beyondsoft.com"
        },
        {
            "github_username": "<redacted>-me",
            "signatureModified": "2019-11-04T03:21:23.444623+0000",
            "signature_id": "8f299af2-da97-4fbd-afb7-278a2f3be929",
            "signed_on": "2019-11-04T03:21:23.444616+0000",
            "user_email": "<redacted>@<redacted>.me",
            "user_name": "<redacted>"
        },
        {
            "github_username": "<redacted>",
            "signatureModified": "2019-10-14T21:37:10.364655+0000",
            "signature_id": "413a813d-5203-4dfa-867c-d00d0c49e20d",
            "signed_on": "2019-10-14T21:37:10.364647+0000",
            "user_email": "ken.<redacted>@gmail.com",
            "user_name": "Ken <redacted>"
        },
        {
            "github_username": "<redacted>",
            "signatureModified": "2020-02-26T20:42:32.547898+0000",
            "signature_id": "ccb88dbb-9ac4-45ab-8682-80e422398f48",
            "signed_on": "2020-02-26T20:42:32.547890+0000",
            "user_email": "ben@<redacted>.com",
            "user_name": "Ben <redacted>"
        },
        {
            "github_username": "tom-<redacted>",
            "signatureModified": "2020-10-13T11:04:29.905204+0000",
            "signature_id": "6e56259d-e462-4c92-bb59-631e2de60fed",
            "signed_on": "2020-10-13T11:05:53Z",
            "user_email": "<redacted>+tom-<redacted>@users.noreply.github.com",
            "user_name": "Tom <redacted>"
        },
        {
            "github_username": "<redacted>-hu",
            "signatureModified": "2020-03-11T21:35:46.870716+0000",
            "signature_id": "35ecf07d-553e-4b6e-98a4-1afc00d35098",
            "signed_on": "2020-03-11T21:35:46.870708+0000",
            "user_email": "iterator.<redacted>@gmail.com",
            "user_name": "<redacted> Hu"
        },
        {
            "github_username": "<redacted>",
            "signatureModified": "2020-03-28T01:07:21.341099+0000",
            "signature_id": "96a0045e-dc65-4bbe-8427-d908f4c9deff",
            "signed_on": "2020-03-28T01:07:21.341091+0000",
            "user_email": "<redacted>@gmail.com",
            "user_name": "<redacted>"
        },
        {
            "github_username": "<redacted>",
            "signatureModified": "2020-05-13T18:19:23.189305+0000",
            "signature_id": "ae4f9629-6385-4314-82b8-b78984cc1b5c",
            "signed_on": "2020-05-13T18:19:23.189298+0000",
            "user_email": "mark.<redacted>@<redacted>.com",
            "user_name": "Mark <redacted>"
        },
        {
            "github_username": "<redacted>",
            "signatureModified": "2020-10-19T21:03:49.500755+0000",
            "signature_id": "bc799dac-fc31-427d-80df-726e1a13c1ba",
            "signed_on": "2020-10-19T21:05:32Z",
            "user_email": "stephan.<redacted>@yahoo.<redacted>",
            "user_name": "Stephan <redacted>"
        },
        {
            "github_username": "<redacted>",
            "signatureModified": "2020-09-16T09:31:02.111647+0000",
            "signature_id": "577423d4-a310-46c8-a4a9-534227df4211",
            "signed_on": "2020-09-16T09:32:48Z",
            "user_email": "<redacted>@<redacted>.com",
            "user_name": "Mehmet <redacted>"
        },
        {
            "github_username": "<redacted>",
            "signatureModified": "2019-10-11T13:02:06.411435+0000",
            "signature_id": "3a6a8f7d-e11d-4a4d-934e-d70d5db8de4f",
            "signed_on": "2019-10-11T13:02:06.411427+0000",
            "user_email": "nick.<redacted>@<redacted>.com",
            "user_name": "Nick <redacted>"
        },
        {
            "github_username": "<redacted>",
            "signatureModified": "2019-11-21T10:24:33.271268+0000",
            "signature_id": "5ad2cf09-75dc-4ec9-a538-2fb67170c97f",
            "signed_on": "2019-11-21T10:24:33.271260+0000",
            "user_email": "<redacted>@alas.dk",
            "user_name": "Morten <redacted>"
        },
        {
            "github_username": "<redacted>",
            "signatureModified": "2020-08-17T10:09:37.992175+0000",
            "signature_id": "9dd82265-eab2-43a3-af48-30b2d504e2c5",
            "signed_on": "2020-08-17T10:15:13Z",
            "user_email": "<redacted>@gmail.com"
        },
        {
            "github_username": "<redacted>",
            "signatureModified": "2020-12-22T16:53:06.756479+0000",
            "signature_id": "5746a577-3894-4e22-9a0d-07a7e310a92a",
            "signed_on": "2020-12-22T16:53:07Z",
            "user_email": "mail@<redacted>.es",
            "user_name": "Rich <redacted>"
        },
        {
            "github_username": "<redacted>",
            "signatureModified": "2020-02-26T19:45:02.405629+0000",
            "signature_id": "d386c9f5-7f2f-415e-8d1c-e5a6c215e321",
            "signed_on": "2020-02-26T19:45:02.405622+0000",
            "user_email": "<redacted>@gmail.com"
        }
    ]
}
```

**Comment by dealako** [comment]
Query for large projects, e.g. 

```
  Cloud Native Computing Foundation (CNCF) - d8cead54-92b7-48c5-a2c8-b1e295e8f7f1
    Live Project              : True
    SalesForce Projects       :
      ID                      : a0941000002wBz4AAE
        Name                  : Cloud Native Computing Foundation (CNCF)
        Parent ID             : None
        Parent Name           : None
        Enabled Services      : ['CLA']
    Repositories              : 67
    Individual Signatures     : 9,620
    Employee Acknowledgements : 3,408
    Company Signatures        : 606
    CLA Managers              : 831
```

there are nearly 10K ICLAs. As a result the query takes around 35 seconds to fetch.  May consider updating the signature records offline with a migration script to add the missing data.

**Comment by vinod-kadam** [comment]
This is deployed to all environments through https://github.com/communitybridge/easycla/releases/tag/v1.0.137

**Comment by vinod-kadam** [comment]
I verified it on Staging - looks fine.
I tested it for Project PerfTestFoundation & CLA Group PerfTest-ClaGroup1 for which bulk ICLAs were created.
I called API https://api-gw.staging.platform.linuxfoundation.org/cla-service/v4/cla-group/a87b899f-8c82-4f26-8acf-eaa772d048f4/icla/signatures & it returned 1216 entries. I verified it on UI & I can see 1217 entries.
![image.png](https://images.zenhubusercontent.com/5e9e9952b186e1325cf5608a/9bedc186-25f3-40d1-b619-6c624729fe77)
Each page has 5 entries; page 244 has only 1 entry so total entries: 1216
